MatrixSDK/Crypto/MXCryptoV2.swift

```diff
@@ -387,18 +387,39 @@ class MXCryptoV2: NSObject, MXCrypto {
         case .verified:
             // If we want to set verified status, we will manually verify the device,
             // including uploading relevant signatures
+            try? machine.setLocalTrust(userId: machine.userId, deviceId: deviceId, trust: .verified)
             
-            Task {
-                do {
-                    try await machine.verifyDevice(userId: userId, deviceId: deviceId)
-                    log.debug("Successfully marked device as verified")
-                    await MainActor.run {
-                        success?()
+            if (userId == machine.userId) {
+                if (machine.crossSigningStatus().hasSelfSigning) {
+                    // if we can cross sign, upload a new signature for that device
+                    Task {
+                        do {
+                            // This method will always fail if the device belongs to someone else.
+                            // XXX Should update API? and remove the userId?
```

Comment:
For the first part of the commet I think `machine.verifyDevice` will raise an exception and this will go on the `Failed marking device as verified`. 

As for logging out the user entirely, that feels a bit invasive, I wouldn't take that up now